src/lib.rs

```diff
@@ -1129,6 +1101,128 @@ fn uses_named_pipe_jobserver(makeflags: &OsStr) -> bool {
         .contains("--jobserver-auth=fifo:")
 }
 
+/// Find [`CMAKE_SYSTEM_NAME`] based on `$CARGO_CFG_TARGET_OS` if available,
+/// otherwise try to infer it from the given target.
+///
+/// [`CMAKE_SYSTEM_NAME`]: https://cmake.org/cmake/help/latest/variable/CMAKE_SYSTEM_NAME.html
+fn get_system_name(target: &str) -> &'static str {
```

Comment:
If you get bored and want more names see my https://github.com/offscale/osbuild/blob/master/src/distribution.c#L15-L130